### PR TITLE
Centralize app_id handling (continuation)

### PR DESCRIPTION
**Pull request by FeralChild64**
# Description

This is the continuation of PR https://github.com/dosbox-staging/dosbox-staging/pull/3817 by [ReillyBrogan](https://github.com/ReillyBrogan). Changes done by me:
- handled review remarks which were still open
- updated some copyright strings
- improved comment formatting in the `dosbox.h` file

## Original description:

On modern Linux it is preferred to use the fully qualified application ID for naming the desktop, icon, and metainfo files. Add a meson option to capture this variable and set it to `org.dosbox-staging.DOSBoxStaging` which better matches that the homepage is dosbox-staging.org.

Then ensure that we're setting the X11 `resourceclass` and the Wayland app_id to this value. Those values need to match the name of the .desktop file in order for compositors/window managers to correctly associate windows to the appropriate .desktop file (see [here](https://github.com/zed-industries/zed/issues/12707#issuecomment-2168742761) for an explanation I previously wrote on how that works). Now that we're ensuring that both X11 and Wayland should have correct window associations we can drop `StartupWMClass` from the desktop file as it is now unnecessary.

Also, since SDL3 changed how that is done I added some #if statements so that that's ready whenever dosbox-staging is ported to SDL3.

# Manual testing

- compiled the project under _Kubuntu_ running inside _VirtualBox_ (I didn't want to potentially mess with my main system), executed `meson install`, checked that the menu system correctly recognizes icons
- using command `xprop -spy` I've checked the `WM_CLASS` looks correctly

(please consider I'm not a Linux packaging specialist, so my testing was limited)

# Checklist

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: contrib/linux/install-icons.sh [hunk 1]
```diff
@@ -25,14 +25,16 @@ usage() {
...
+    cp desktop/org.dosbox-staging.dosbox-staging.desktop $DEST_PATH/applications/org.dosbox-staging.dosbox-staging.desktop
 
     ## Replace executable location in the .desktop file
-    sed -i 's\Exec=dosbox\Exec='$(realpath dosbox)'\g' $DEST_PATH/applications/dosbox-staging.desktop
+    sed -i 's\Exec=dosbox\Exec='$(realpath dosbox)'\g' $DEST_PATH/applications/org.dosbox-staging.dosbox-staging.desktop
 }
 
 remove_icons() {
     find $DEST_PATH/icons -name dosbox-staging* -delete
+    rm $DEST_PATH/applications/org.dosbox-staging.dosbox-staging.desktop
```
Comment (johnnovak) [inline contrib/linux/install-icons.sh:36]: Maybe extract `org.dosbox-staging.dosbox-staging.desktop` to a var?

**Review by johnnovak — APPROVED** [review]
Looks fine to me @FeralChild64

File: contrib/linux/install-icons.sh [hunk 2]
```diff
@@ -25,14 +25,16 @@ usage() {
...
-    cp desktop/dosbox-staging.desktop $DEST_PATH/applications
+    cp desktop/org.dosbox-staging.dosbox-staging.desktop $DEST_PATH/applications/org.dosbox-staging.dosbox-staging.desktop
 
     ## Replace executable location in the .desktop file
-    sed -i 's\Exec=dosbox\Exec='$(realpath dosbox)'\g' $DEST_PATH/applications/dosbox-staging.desktop
+    sed -i 's\Exec=dosbox\Exec='$(realpath dosbox)'\g' $DEST_PATH/applications/org.dosbox-staging.dosbox-staging.desktop
 }
 
 remove_icons() {
     find $DEST_PATH/icons -name dosbox-staging* -delete
```
Comment (weirddan455) [inline contrib/linux/install-icons.sh:35]: Same comment I made on the original PR.  This no longer deletes icons since the names have changed.

Comment (johnnovak) [reply contrib/linux/install-icons.sh:35]: Fair point.

File: contrib/linux/install-icons.sh (hunk 1 above)
Comment (FeralChild64) [reply contrib/linux/install-icons.sh]: OK, changed.

File: contrib/linux/install-icons.sh (hunk 2 above)
Comment (FeralChild64) [reply contrib/linux/install-icons.sh:36]: I think I misunderstood the comment - to fix the problem I have added lines:

```bash
    ## Also remove legacy files, if present
    rm $DEST_PATH/applications/dosbox-staging.desktop
```

to the `remove_icons()` function. AFAIK this `sed` command does not remove eny files, it just updates the content of the installed `.desktop` file. Shall I remove legacy files here, too? I don't really know what are the exactl use cases of this script, unfortunately.

Comment (weirddan455) [reply contrib/linux/install-icons.sh:36]: The script is meant to be run by end users that download the tarball from our website.  It copies the icon files and .desktop file to either `~/.local/share/` or, with the system flag `/usr/share/applications`.  This installs a shortcut and is also needed by Wayland to show the Staging icon in the task bar (Wayland doesn't have a way to programmatically set this unlike X11 so it looks for a .desktop file that matches the executable).

These files are re-named in this PR so the remove function will not work.  You've got the .desktop part handled correctly but not the .png files.

Comment (FeralChild64) [reply contrib/linux/install-icons.sh]: OK, now I understand. Fixed.

**Review by weirddan455 — APPROVED** [review]
This looks good to me now as far as I can tell.

**Review by interloper98 — APPROVED** [review]
It's too bad we can't use a common list + for loop to grind through all the resolutions, but it seems every OS and desktop cares about a slight different set! Oh well.
 
No concerns from me; looks simple and maintainable. If it helps more Linux users get on board, then that's great.

**Review by johnnovak — APPROVED** [review]
Thanks @FeralChild64 @interloper98 and @weirddan455 , merging.

**Comment by yselkowitz** [comment]
The problem with this name is that it is not "legal" in some uses, particularly flatpaks:

https://github.com/flatpak/flatpak-builder/issues/37
https://specifications.freedesktop.org/desktop-entry-spec/latest/file-naming.html
https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names
https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic

**Comment by FeralChild64** [comment]
@yselkowitz OK, thanks for the info, I’ll look into this.

**Pull request merged**